### ISSUE TITLE
Retire `DataFrame.to_hdf`

### ISSUE THREAD

**Issue by hendrikmakait** [dask/dask#10131]
As suggested by @mrocklin in https://github.com/dask/distributed/pull/7720#issuecomment-1488800571, we should retire `DataFrame.to_hdf` as legacy storage format that causes some issues during maintenance.

**Open question**
Should we remove more HDF-related code?

**Comment by TomAugspurger** [comment]
We still get comments about the msgspec removal from a few years ago, and
I think that was never as widely used as hdf5 🙂

But that said, HDF5 and dask.dataframe have always had some challenges.

https://examples.dask.org/surveys/2021.html#What-file-formats-do-you-typically-work-with?
has
a bit of data here (HDF5 was higher than I expected, FWIW).

**Comment by phofl** [comment]
I don't think that we have any direct data, but FWIW we (pandas) barely get bug-reports/enhancement requests for HDF on our issue tracker

**Comment by mrocklin** [comment]
The HDF5 responses on the Dask survey are likely due to dask.array

My question was more about Pandas users.  If @phofl is saying that HDF doesn't come up often in that broader audience then I'm more than happy to retire it in dask.dataframe where I expect Parquet will be even more the norm.
